mu-plugins/blocks/latest-news/latest-news.php

```diff
@@ -33,8 +33,10 @@ function render_block( $attributes ) {
 
 	// Check if we can switch to the News blog.
 	// @todo Prevent switch if Rosetta, Rosetta should use local posts.
-	if ( should_switch_to_blog() && defined( 'WPORG_NEWS_BLOGID' ) ) {
-		switch_to_blog( WPORG_NEWS_BLOGID );
+	if ( should_switch_to_blog() && ( defined( 'WPORG_NEWS_BLOGID' ) || isset( $attributes['blogId'] ) ) ) {
+		// Prefer the attribute blogId over the constant.
+		$blog_id = isset( $attributes['blogId'] ) ? $attributes['blogId'] : WPORG_NEWS_BLOGID;
```

Comment:
Right, It was originally undefined but that created some react issues with controlled/uncontrolled inputs. I'll check for `0`.

Reply:
I haven't spun anything up to test it, but will you need to update the block on the homepage when this is merged? Or will it still default to the News blog?

Reply:
We'll need to update the block on the [homepage](https://github.com/WordPress/wporg-main-2022/blob/d3fc83080ed67056133537c2229da69619f12db2/source/wp-content/themes/wporg-main-2022/patterns/home.php#L297). We could alternatively just default it to `8` but having no default probably makes sense. It seems to be showing up more in different contexts. 

Since we now check that it isn't `0` it should fix the "no post found" issues as it won't switch blogs.